ampule.py

```diff
@@ -80,7 +80,22 @@ def __send_response(client, code, headers, data):
         response += "%s: %s\r\n" % (k, v)
     response += "\r\n" + data + "\r\n"
 
-    client.send(response)
+    # unreliable sockets on ESP32-S2: see https://github.com/adafruit/circuitpython/issues/4420#issuecomment-814695753
+    response_length = len(response)
+    bytes_sent_total = 0
+    while True:
+        try:
+            bytes_sent_total += client.send(response)
+            if bytes_sent_total >= response_length:
+                return bytes_sent_total
+            else:
+                response = response[bytes_sent:]
```

Comment:
your modification won't work. you got rid of the bytes_sent var but would've needed it here